de-lib/src/main/java/org/iplantc/de/diskResource/client/presenters/grid/GridViewPresenterImpl.java

```diff
@@ -483,8 +499,55 @@ public void onSuccess(ShareResourceLinkDialog result) {
                 }
             });
         } else {
-            diskResourceService.createDataLinks(Arrays.asList(toBeShared.getPath()),
-                                                new CreateDataLinksCallback());
+            diskResourceService.listDataLinks(diskResourceUtil.asStringPathList(Arrays.asList(toBeShared)),
+                                              new DataCallback<String>() {
+
+                                                  @Override
+                                                  public void onFailure(Integer statusCode,
+                                                                        Throwable exception) {
+                                                      ErrorHandler.post(appearance.createDataLinksError(),
+                                                                        exception);
+                                                  }
+
+                                                  @Override
+                                                  public void onSuccess(String result) {
+                                                      List<DataLink> dlList = processResult(result);
+                                                      if (dlList == null || dlList.isEmpty()) {
+                                                          diskResourceService.createDataLinks(Arrays.asList(
+                                                                  toBeShared.getPath()),
+                                                                                              new CreateDataLinksCallback());
+                                                      } else {
+                                                          showPublicLink(dlList);
+                                                      }
+
+                                                  }
+
+                                                  private List<DataLink> processResult(String result) {
```

Comment:
Probably not for this PR, but it looks like at least some of the logic in `processResult` is also in `ListDataLinksCallback`.  It might be nice to refactor this so the logic isn't duplicated, maybe in a callback converter or something.